skcosmo/sample_selection/_base.py

```diff
@@ -346,3 +397,226 @@ def __init__(
             full=full,
             random_state=random_state,
         )
+
+
+class DirectionalConvexHull:
+    """
+    Performs Sample Selection by constructing a Directional Convex Hull and determining the distance to the hull as outlined in the reference
+
+    Parameters
+    ----------
+
+    low_dim_idx    : list of ints, default None
+                   Indices of columns of X containing features to be used for the
+                   directional convex hull construction (also known as the low-
+                   dimensional (LD) hull). By default [0] is used.
+
+    Attributes
+    ----------
+
+    high_dim_idx_   : list of ints
+                    Indices of columns in data containing high-dimensional
+                    features (i.e. those not used for the convex hull
+                    construction)
+
+    vertices_idx_ : numpy.ndarray
+                    Indices of datapoints that form the vertices of the
+                    convex hull
+    interpolator_high_dim_  : scipy.interpolate.interpnd.LinearNDInterpolator
+                    Interpolater for the features in the high-
+                    dimensional space
+
+    interpolator_y_  : scipy.interpolate.interpnd.LinearNDInterpolator
+                    Interpolater for the targets y
+
+    References
+    ----------
+    .. [dch] A. Anelli, E. A. Engel, C. J. Pickard and M. Ceriotti,
+             Physical Review Materials, 2018.
+    """
+
+    def __init__(self, low_dim_idx=None):
+        self.low_dim_idx = low_dim_idx
+
+        if low_dim_idx is None:
+            self.low_dim_idx = [0]
+        else:
+            self.low_dim_idx = low_dim_idx
+
+    def fit(self, X, y):
+        """
+        Learn the samples that form the convex hull.
+
+        Parameters
+        ----------
+        X        : ndarray of shape (n_samples, n_features)
+                   Feature matrix of samples to use for constructing the convex
+                   hull.
+        y        : ndarray of shape (n_samples,)
+                   Target values (property on which the convex hull should be
+                   constructed, e.g. Gibbs free energy)
+
+        Returns
+        -------
+        self : object
+            Fitted scorer.
+        """
+
+        X, y = check_X_y(X, y, ensure_min_features=2)
+        self.n_features_in_ = X.shape[1]
+
+        if len(y.shape) == 1:
+            y = y.reshape((len(y), 1))
```

Comment:
Do we want this to work for `y.shape[1]!=0`? If not there should be a check here

Reply:
the case y.shape = (n_samples, 1) is allowed
the case y.shape = (n_samples, >1)  is checked by the check_X_y (argument multioutput is False by default) 

I put the flag into the check to make it more apparent